src/main/java/runhistoryplus/patches/PotionUseAndDiscardRunHistoryPatch.java

```diff
@@ -179,4 +251,98 @@ public int[] Locate(CtBehavior ctMethodToPatch) throws CannotCompileException, P
             }
         }
     }
+
+    public static void obtainPotionAlchemizeOrEntropicBrew(final AbstractPotion potion, String source) {
+        switch(source) {
```

Comment:
With the enum approach I suggested in another comment, you'll need to add an `if (source != null)` check surrounding this switch block.